[Threat Hunting Investigations] Fix timeline column width bug

janmonschke opened the pull request:
## Summary

Fixes: https://github.com/elastic/kibana/issues/213754

The issue above describes a bug in timeline that makes it impossible to change the width of a timeline column. This PR fixes that issue and makes sure that timeline column width settings are saved to localStorage. This mimics the behaviour of the alerts table elsewhere in security solution.

https://github.com/user-attachments/assets/8b9803a0-406d-4f2d-ada5-4c0b76cd6ab8



<!--ONMERGE {"backportTargets":["7.17","8.16","8.17","8.18","8.x","9.0"]} ONMERGE-->

janmonschke committed:
forward timeline column settings to UnifiedDataTable

janmonschke committed:
merge onColumnResize and onResizeDataGrid since onColumnResize wasn't used anywhere else

janmonschke committed:
store and read timeline column config from localStorage

janmonschke committed:
add tests

janmonschke committed:
rename tests

janmonschke committed:
remove nasty type cast

elasticmachine commented:
Pinging @elastic/security-threat-hunting-investigations (Team:Threat Hunting:Investigations)

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/283305)
* Commit: 23276e5e3684e7b68aa977ccfb8f00f93644361f

### Failed CI Steps
* [Osquery Cypress Tests #7](https://buildkite.com/elastic/kibana-pull-request/builds/283305#01958ac8-7296-40dd-b43b-8b7db9d975f0)

cc @janmonschke

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/283305","number":283305,"commit":"23276e5e3684e7b68aa977ccfb8f00f93644361f"}],"number":283305}
buildkite-pr-comment-->

lgestc reviewed (approved):
looks like a proper fix to me, it even adds the persistence 🎉

janmonschke commented:
@elasticmachine merge upstream

elasticmachine committed:
Merge branch 'main' into security/timeline-column-width-bug

elasticmachine commented:
## :yellow_heart: Build succeeded, but was flaky
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/283424)
* Commit: 7596217d79db1ef06f355472a886e557ead142c0

### Failed CI Steps
* [FTR Configs #14](https://buildkite.com/elastic/kibana-pull-request/builds/283424#01958b9a-221c-458d-9fd3-ca28447a9bae)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/283424#01958b9a-221c-458d-9fd3-ca28447a9bae) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/283424/jobs/01958b9a-221c-458d-9fd3-ca28447a9bae/artifacts/01958bc0-9266-46e0-8e01-3742a2227f67) FTR Configs #<span></span>14 / Cloud Security Posture GET /internal/cloud_security_posture/stats CSPM Compliance Dashboard Stats API should return CSPM benchmarks V2 

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/07012811b29b487a3b4a664469c7a198355e44bf) | [after](https://github.com/elastic/kibana/commit/7596217d79db1ef06f355472a886e557ead142c0) | diff |
| --- | --- | --- | --- |
| `securitySolution` | 7078 | 7079 | +1 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/07012811b29b487a3b4a664469c7a198355e44bf) | [after](https://github.com/elastic/kibana/commit/7596217d79db1ef06f355472a886e557ead142c0) | diff |
| --- | --- | --- | --- |
| `securitySolution` | 8.9MB | 8.9MB | +896.0B |



### History
* :broken_heart: [Build #283305](https://buildkite.com/elastic/kibana-pull-request/builds/283305) failed 23276e5e3684e7b68aa977ccfb8f00f93644361f

cc @janmonschke

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/283424","number":283424,"commit":"7596217d79db1ef06f355472a886e557ead142c0"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/283305","number":283305,"commit":"23276e5e3684e7b68aa977ccfb8f00f93644361f"}],"number":283424}
buildkite-pr-comment-->

janmonschke committed:
apply column config also to new timelines

christineweng reviewed (commented):
Resizing fix LGTM, thanks for putting in the fix! 

Some observations during desk testing:

- The column width does not persist when a column is removed. In this video, I first resize `message` to be quite narrow, upon removing another column, the message width resets

https://github.com/user-attachments/assets/cde4d10d-6fc2-4af1-b1c7-1624d3b2647b

- Some fields align to the right, it can be hard to find where the resize area is. It appears to impact only certain field types such as number and ip
![image](https://github.com/user-attachments/assets/68aedcf2-dd21-435f-84f2-0f2c4c4f7f96)



- The alignment leads to this shifting mechanism, it gets annoying quickly when 2 fields are like that side by side


https://github.com/user-attachments/assets/75222a23-f482-4a6a-82af-ca1eea435f59

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/283469)
* Commit: 2cfadfb259f74203faf49cd2b769cb692648884f

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/07012811b29b487a3b4a664469c7a198355e44bf) | [after](https://github.com/elastic/kibana/commit/2cfadfb259f74203faf49cd2b769cb692648884f) | diff |
| --- | --- | --- | --- |
| `securitySolution` | 7078 | 7079 | +1 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/07012811b29b487a3b4a664469c7a198355e44bf) | [after](https://github.com/elastic/kibana/commit/2cfadfb259f74203faf49cd2b769cb692648884f) | diff |
| --- | --- | --- | --- |
| `securitySolution` | 8.9MB | 8.9MB | +938.0B |



### History
* :yellow_heart: [Build #283424](https://buildkite.com/elastic/kibana-pull-request/builds/283424) was flaky 7596217d79db1ef06f355472a886e557ead142c0
* :broken_heart: [Build #283305](https://buildkite.com/elastic/kibana-pull-request/builds/283305) failed 23276e5e3684e7b68aa977ccfb8f00f93644361f

cc @janmonschke

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/283469","number":283469,"commit":"2cfadfb259f74203faf49cd2b769cb692648884f"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/283424","number":283424,"commit":"7596217d79db1ef06f355472a886e557ead142c0"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/283305","number":283305,"commit":"23276e5e3684e7b68aa977ccfb8f00f93644361f"}],"number":283469}
buildkite-pr-comment-->

janmonschke committed:
remove unused `upsertTimelineColumn` and change tests to use the new `upsertTableColumn`

janmonschke committed:
remove unused helper `applyDeltaToTimelineColumnWidth`

janmonschke committed:
make sure columns always contain the local config

janmonschke commented:
> The column width does not persist when a column is removed. In this video, I first resize message to be quite narrow, upon removing another column, the message width resets

@christineweng Nice catch, I fixed this behaviour in the recent commits and added tests for these cases. Could you have another look at it? :)

janmonschke commented:
> Some fields align to the right, it can be hard to find where the resize area is. It appears to impact only certain field types such as number and ip
@christineweng yeah, this comes from the underlying EUI component. Not much I can do at this point ;)

logeekal commented:
>The alignment leads to this shifting mechanism, it gets annoying quickly when 2 fields are like that side by side

I also do not know reasoning behind it. but field align based on the type. For example, text aligns to the left, numbers align to right, etc

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/283591)
* Commit: 8b5b119f16017613de6d61c472c4eb56badcbec9

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/07012811b29b487a3b4a664469c7a198355e44bf) | [after](https://github.com/elastic/kibana/commit/8b5b119f16017613de6d61c472c4eb56badcbec9) | diff |
| --- | --- | --- | --- |
| `securitySolution` | 7078 | 7079 | +1 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/07012811b29b487a3b4a664469c7a198355e44bf) | [after](https://github.com/elastic/kibana/commit/8b5b119f16017613de6d61c472c4eb56badcbec9) | diff |
| --- | --- | --- | --- |
| `securitySolution` | 8.9MB | 8.9MB | +947.0B |



### History
* :green_heart: [Build #283469](https://buildkite.com/elastic/kibana-pull-request/builds/283469) succeeded 2cfadfb259f74203faf49cd2b769cb692648884f
* :yellow_heart: [Build #283424](https://buildkite.com/elastic/kibana-pull-request/builds/283424) was flaky 7596217d79db1ef06f355472a886e557ead142c0
* :broken_heart: [Build #283305](https://buildkite.com/elastic/kibana-pull-request/builds/283305) failed 23276e5e3684e7b68aa977ccfb8f00f93644361f

cc @janmonschke

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/283591","number":283591,"commit":"8b5b119f16017613de6d61c472c4eb56badcbec9"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/283469","number":283469,"commit":"2cfadfb259f74203faf49cd2b769cb692648884f"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/283424","number":283424,"commit":"7596217d79db1ef06f355472a886e557ead142c0"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/283305","number":283305,"commit":"23276e5e3684e7b68aa977ccfb8f00f93644361f"}],"number":283591}
buildkite-pr-comment-->

janmonschke committed:
clean up naming and unused methods

janmonschke committed:
reset happens globally in `beforeEach`

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/283681)
* Commit: a1b34f474bf998aec8a0a9d6c1f76a251c23b690

### Failed CI Steps
* [Serverless Investigations - Security Solution Cypress Tests #3](https://buildkite.com/elastic/kibana-pull-request/builds/283681#01958f90-559f-46e2-8536-cb832f77da13)

### History
* :green_heart: [Build #283591](https://buildkite.com/elastic/kibana-pull-request/builds/283591) succeeded 8b5b119f16017613de6d61c472c4eb56badcbec9
* :green_heart: [Build #283469](https://buildkite.com/elastic/kibana-pull-request/builds/283469) succeeded 2cfadfb259f74203faf49cd2b769cb692648884f
* :yellow_heart: [Build #283424](https://buildkite.com/elastic/kibana-pull-request/builds/283424) was flaky 7596217d79db1ef06f355472a886e557ead142c0
* :broken_heart: [Build #283305](https://buildkite.com/elastic/kibana-pull-request/builds/283305) failed 23276e5e3684e7b68aa977ccfb8f00f93644361f

cc @janmonschke

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/283681","number":283681,"commit":"a1b34f474bf998aec8a0a9d6c1f76a251c23b690"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/283591","number":283591,"commit":"8b5b119f16017613de6d61c472c4eb56badcbec9"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/283469","number":283469,"commit":"2cfadfb259f74203faf49cd2b769cb692648884f"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/283424","number":283424,"commit":"7596217d79db1ef06f355472a886e557ead142c0"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/283305","number":283305,"commit":"23276e5e3684e7b68aa977ccfb8f00f93644361f"}],"number":283681}
buildkite-pr-comment-->

kibanamachine commented:
Starting backport for target branches: 7.17, 8.16, 8.17, 8.18, 8.x, 9.0

https://github.com/elastic/kibana/actions/runs/13842370576

kibanamachine commented:
## 💔 Some backports could not be created

| Status | Branch | Result |
|:------:|:------:|:------|
|❌|7.17|Backport failed because of merge conflicts|
|❌|8.16|Backport failed because of merge conflicts|
|❌|8.17|Backport failed because of merge conflicts|
|✅|8.18|[<img src="https://img.shields.io/github/pulls/detail/state/elastic/kibana/214471">](https://github.com/elastic/kibana/pull/214471)|
|✅|8.x|[<img src="https://img.shields.io/github/pulls/detail/state/elastic/kibana/214472">](https://github.com/elastic/kibana/pull/214472)|
|✅|9.0|[<img src="https://img.shields.io/github/pulls/detail/state/elastic/kibana/214473">](https://github.com/elastic/kibana/pull/214473)|

Note: Successful backport PRs will be merged automatically after passing CI.

### Manual backport
To create the backport manually run:
```
node scripts/backport --pr 214178
```

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sorenlouv/backport)

<!--- Backport version: 9.6.6 -->

janmonschke commented:
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.17|[<img src="https://img.shields.io/github/pulls/detail/state/elastic/kibana/214519">](https://github.com/elastic/kibana/pull/214519)|
|✅|8.16|[<img src="https://img.shields.io/github/pulls/detail/state/elastic/kibana/214520">](https://github.com/elastic/kibana/pull/214520)|

Note: Successful backport PRs will be merged automatically after passing CI.

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sorenlouv/backport)

<!--- Backport version: 9.6.6 -->
